### PR TITLE
Add API docs

### PR DESCRIPTION
**Pull request by kcpevey**
Fixes | Closes | Resolves #

## Changes introduced in this PR:

- add api doc md file
- add script to generate md file
- add gh workflow to autogenerate docs

## Types of changes

What types of changes does your PR introduce?
- [x] Documentation Update

### Requires testing

- [ ] Yes
- [x] No

### In case you checked yes, did you write tests?

- [ ] Yes
- [ ] No

## Documentation

Does your contribution include breaking changes or deprecations?
If so have you updated the documentation?

- [ ] Yes, docstrings
- [ ] Yes, main documentation
- [ ] Yes, deprecation notices

## Further comments (optional)

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered and more.

### CODE REVIEW

**Comment by kcpevey** [comment]
There is a sister PR on the nebari-docs repo to add it this actual file to the documentation. I still need to add a separate GH action to open the PR on the nebari-docs repo

**Comment by costrouc** [comment]
> There is a sister PR on the nebari-docs repo to add it this actual file to the documentation. I still need to add a separate GH action to open the PR on the nebari-docs repo

@kcpevey  is this PR ready for review or testing? Looks like you said it depends on nebari-docs?

**Comment by kcpevey** [comment]
@costrouc this is ready for review.

File: docs/api_docs/api_doc_mods.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    for kidx in keep_idx:
+        offset_idx.append(id_idx[id_idx.index(kidx) + 1])
+
+    all_headers = zip(keep_idx, offset_idx)
+    map = dict(zip(keep_idx, all_headers))
+
+    # for idx in keep_idx:
+    for idx, group in map.items():
+        clean_list.extend(text[slice(*group)])
+    print(Path(outpath).resolve())
```
Comment (kcpevey) [inline docs/api_docs/api_doc_mods.py:94]: need to fix this

File: pyproject.toml [hunk 2]
```diff
@@ -82,6 +82,11 @@ dev = [
     "pytest",
     "pytest-timeout",
 ]
+docs = [
+    "novella",
+    "pydoc-markdown",
+    "pre-commit"
```
Comment (kcpevey) [inline pyproject.toml:88]:
```suggestion
    "pre-commit",
```

**Comment by kcpevey** [comment]
@trallard this is still failing to find the secret called `NEBARI_SENSEI_API_DOCS_PR_OPENER`. Has it been added here? I dont have permissions to check.

**Comment by aktech** [comment]
The secret is already added, but it won't be accessible by actions run by PR from forks. We would need to trigger via [contributor tests](https://github.com/nebari-dev/nebari/blob/e4401b248e25fcaf901b87fb9f3835fe192f9d8c/.github/workflows/contributor.yaml):

Screenshot 2023-01-24 at 6 30 10 pm

**Comment by kcpevey** [comment]
Thanks @aktech. I'll look into that.

**Comment by kcpevey** [comment]
This is being superseded by https://github.com/nebari-dev/nebari/pull/1634

**Pull request closed without merging**